### PR TITLE
Add spectator mode toggle

### PR DESCRIPTION
**Pull request by iEli2tyree011**

### CODE REVIEW

**Comment by TiTo-418** [comment]
The diamond (in spectator mode) only works if you right-click in a block

**Comment by Pablete1234** [comment]
To get into spectator mode both right and left click work
To get out of spectator mode either left click, or right click against a block, right clicking air doesn't work

**Comment by Pablete1234** [comment]
@iEli2tyree011 it would be nice if right click against chests, furnaces, or other stuff didn't change you back to regular spectator (atm it both opens it and gets you in gm 1)

**Comment by twizmwazin** [comment]
Marc pointed out that there are actual right-click functions in spectator mode, it may be better to limit it to left clicking to change back to creative.

**Comment by Pablete1234** [comment]
right click opens inventories that you can't edit, and left click gets you inside players

**Comment by TiTo-418** [comment]
You need to be in gamemode1 and leftclick on a player to go inside the player

**Comment by Pablete1234** [comment]
I think how it should work:
  right click should be ignored
  left click: 
    if in gm 1, change to gm 3
    if in gm 3, and not pointing towards a player, change to gm 1
it's kinda annoying that to ride a player you have to be in gm1, and left click (gets you in gm3 and ridding the other player)

**Comment by TiTo-418** [comment]
@Pablete1234 Following you logic right click should be ignored only when you're not pointing to a entity block

**Comment by Pablete1234** [comment]
@TiTo-418 right click should be ignored bcos it doesn't work properly, you cant right click out of gm 3 if you are not pointing towards a block, it makes no sense that you can use right click to get in, but not to get out, so just remove it, and leave right click for inventory-opening

File: src/main/java/in/twizmwaz/cardinal/module/modules/respawn/RespawnModule.java [hunk 1]
```diff
@@ -235,4 +209,19 @@ public void onTeamChange(PlayerChangeTeamEvent event) {
 
     }
 
+    public void giveObserversKit(Player player) {
+        player.getInventory().setItem(0, new ItemStack(Material.COMPASS));
```
Comment (Alfalik) [inline src/main/java/in/twizmwaz/cardinal/module/modules/respawn/RespawnModule.java]: Could you "integrate" #768 here?

**Comment by TiTo-418** [comment]
@iEli2tyree011 This PR should be rebased

**Comment by Pablete1234** [comment]
@iEli2tyree011 as @TiTo-418 needs to be rebased, and i think joined players should be able to toggle spec mode before/after the match starts

**Comment by TiTo-418** [comment]
Any plans to merge this?

**Comment by Pablete1234** [comment]
@TiTo-418 it's not ready to merge

Any plans on finnishing this?

**Comment by TiTo-418** [comment]
Implemented in https://github.com/twizmwazin/CardinalPGM/commit/56b4c5b5896387ad579ea0de62f7f823b0fba813

**Comment by Pablete1234** [comment]
@TiTo-418 only part of this PR has been added, the spectator toggle is still part of this PR

**Pull request closed without merging**